### PR TITLE
Azure 161092666

### PR DESCRIPTION
**Pull request by carlo-colombo**
Azure documentation

it should cover

* How to install Ops Manager
* How to configure Ops Manager for PKS
* How to install the PKS tile
* How to configure the PKS tile for Azure
* How to configure the Load Balancer for the PKS API
* How to configure PV storage classes for Azure (dependant on these working)
* How to configure a Load Balancer for the cluster

[#161218943]

### CODE REVIEW

File: azure-api-load-balancer.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+title: Creating an Azure Load Balancer for the PKS API
+owner: PKS
+---
+
+<strong><%= modified_date %></strong>
+
+This topic describes how to create a load balancer for the PKS API using Azure.
+
+Before you install PKS, you must configure an external TCP load balancer to
```
Comment (bstick12) [inline azure-api-load-balancer.html.md.erb]: @carlo-colombo This page details that the load balancer should be setup before installing PKS API. Some of the language further down infers that the PKS-API has already been installed. Can you clarify

File: azure-om-config.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+	* **Subscription ID**: Enter the ID of your Azure subscription.
+	* **Tenant ID**: Enter your `TENANT_ID`.
+	* **Application ID**: Enter the `APPLICATION_ID` that you created in the [Create an Azure Active Directory Application](./prepare-env-manual.html#create-aad) step of the _Preparing to Deploy PCF on Azure_ topic.
+	* **Client Secret**: Enter your `CLIENT_SECRET`.
+
+1. Complete the following fields:
+	* **Resource Group Name**: Enter the name of your resource group, which is exported from Terraform as the output `pcf_resource_group_name`.
+	* **BOSH Storage Account Name**: Enter the name of your storage account, which is exported from Terraform as the output `bosh_root_storage_account`.
+
+1. For **Cloud Storage Type**, select **Use Storage Accounts** and enter the `wildcard_vm_storage_account` output from Terraform.
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo Populating this as recommended gave the following error 

```
''*ecuboshvms*' does not match any Storage Accounts under Resource Group 'ecu'', type: IaasConfigurationVerifier

```

File: azure-om-config.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+    <%= image_tag("images/director-tile-azure.png") %>
+
+1. Select **Azure Config**.
+    <%= image_tag("images/azure-config.png") %>
+
+1. Complete the following fields with information you obtained in the [Preparing to Deploy PCF on Azure](./prepare-env-manual.html) topic.
+	* **Subscription ID**: Enter the ID of your Azure subscription.
+	* **Tenant ID**: Enter your `TENANT_ID`.
+	* **Application ID**: Enter the `APPLICATION_ID` that you created in the [Create an Azure Active Directory Application](./prepare-env-manual.html#create-aad) step of the _Preparing to Deploy PCF on Azure_ topic.
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo This link doesn't work. 

We should document that the `APPLICATION_ID` is the same as the `CLIENT_ID`

File: azure-om-config.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+1. Click **Save**.
+
+## <a id='director-config'></a>Step 3: Director Config Page ##
+
+1. Select **Director Config** to open the **Director Config** page.
+
+    <%= image_tag("aws/director-aws.png") %>
+
+1. Enter at least two of the following NTP servers in the **NTP Servers (comma
+delimited)** field, separated by a comma: `0.amazon.pool.ntp.org,1.amazon.pool.ntp.org,2.amazon.pool.ntp.org,3.amazon.pool.ntp.org`
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo Should we be using amazon ntp servers here

File: azure-om-config.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+1. Select **Create Networks**.
+<%= image_tag("aws/create-networks.png") %>
+
+1. (Optional) Select **Enable ICMP checks** to enable ICMP on your networks. The BOSH Director uses ICMP checks to confirm that components within your network are reachable.
+
+1. Perform the following steps to add the network configuration you created for your VPC:
+
+  1. Click **Add Network**.
+
+  1. For **Name**, enter `pks-infrastructure-network`.
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo I had  to enter `ecu-management-subnet`

File: azure-om-config.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+  1. For each availability zone, click **Add Subnet** and use the information in the following table to create three subnets.
+  <table>
+    <tr>
+      <th rowspan="7">First Subnet</th>
+      <th>Field</th>
+      <th>Configuration</th>
+    </tr>
+    <tr>
+      <td>VPC Subnet ID</td>
+      <td>The first value of <code>infrastructure_subnet_ids</code> from the <code>terraform output</code>.</td>
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo Can we improve this as you need to have the resource group/subnet name for Azure. e.g. `ecu-virtual-network/ecu-infrastructure-subnet`

File: azure-om-config.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+      <td>CIDR</td>
+      <td><code>10.0.8.0/26</code></td>
+    </tr>
+    <tr>
+      <td>Reserved IP Ranges</td>
+      <td><code>10.0.8.0 - 10.0.8.9</code></td>
+    </tr>
+    <tr>
+      <td>DNS</td>
+      <td><code>168.63.129.16</code><sup><strong>*</strong></sup></td>
```
Comment (bstick12) [inline azure-om-config.html.md.erb]: @carlo-colombo What is this IP address?

File: azure-api-load-balancer.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,99 @@
+---
+title: Creating an Azure Load Balancer for the PKS API
+owner: PKS
+---
+
+<strong><%= modified_date %></strong>
+
+This topic describes how to create a load balancer for the PKS API using Azure.
```
Comment (bstick12) [inline azure-api-load-balancer.html.md.erb]: @skelly-pivotal @carlo-colombo Can we look at getting the latest Azure terraform up and working as this LB is created by the new terraform and would massively simplify an area that is always problematic

File: installing-pks-azure.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+1. Enter **Resource Group**, which is the name of the resource group that the cluster is deployed in.
+
+1. Enter **Virtual Network**, which is the name of the virtual network that the cluster is deployed in.
+
+1. Enter **Virtual Network Resource Group**, which is the name of the resource group that the virtual network is deployed in.
+
+1. Enter **Default Security Group**, which is the name of the security group attached to the cluster's subnet.
+
+1. Enter **Primary Availability Set**, which is the name of the availability set that should be used as the load balancer backend.
```
Comment (bstick12) [inline installing-pks-azure.html.md.erb]: @carlo-colombo Where do I get this information and how was the availability set created

File: installing-pks-azure.html.md.erb [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+1. Enter **Location**, which is the location of the resource group that the cluster is deployed in (eg. northeurope).
+
+1. Enter **Resource Group**, which is the name of the resource group that the cluster is deployed in.
+
+1. Enter **Virtual Network**, which is the name of the virtual network that the cluster is deployed in.
+
+1. Enter **Virtual Network Resource Group**, which is the name of the resource group that the virtual network is deployed in.
+
+1. Enter **Default Security Group**, which is the name of the security group attached to the cluster's subnet.
```
Comment (bstick12) [inline installing-pks-azure.html.md.erb]: @carlo-colombo Where do I get this information and how was the availability set created

File: installing-pks-azure.html.md.erb (hunk 9 above)
Comment (bstick12) [reply installing-pks-azure.html.md.erb]: @carlo-colombo Looks like we create it in our environments using https://github.com/pivotal-cf/pks-home/blob/master/azure/availability_set.tf but that isn't in the Azure Terraform

File: azure-om-config.html.md.erb (hunk 2 above)
Comment (carlo-colombo) [reply azure-om-config.html.md.erb]: Changed to use `bosh_storage_account_name`

File: azure-api-load-balancer.html.md.erb (hunk 1 above)
Comment (carlo-colombo) [reply azure-api-load-balancer.html.md.erb]: I think all of this is outdated with the newer version of terraform (once they add the 9021 port to the load balancer), we can just point to the load balancer in the install-pks-azure page.

File: azure-om-config.html.md.erb (hunk 3 above)
Comment (carlo-colombo) [reply azure-om-config.html.md.erb]: Do you mean for the k8s cloud provider?

File: azure-om-config.html.md.erb (hunk 4 above)
Comment (carlo-colombo) [reply azure-om-config.html.md.erb]: fixed, we copied this (and all the om-azure-config) form PAS documentation. We use `0.pool.ntp.org,1.pool.ntp.org,2.pool.ntp.org,3.pool.ntp.org` in our scripts :-)

File: azure-om-config.html.md.erb (hunk 7 above)
Comment (bstick12) [reply azure-om-config.html.md.erb]: Looks to be an Azure DNS address - https://blogs.msdn.microsoft.com/mast/2015/05/18/what-is-the-ip-address-168-63-129-16/

File: azure-api-load-balancer.html.md.erb (hunk 8 above)
Comment (carlo-colombo) [reply azure-api-load-balancer.html.md.erb]: I tried today while setting up the toi, the load balancer is missing a few things at the moment (load balancing rule on 9021, and the security group does not seems set up) but I agree on using the terraforming lb

File: azure-om-config.html.md.erb (hunk 6 above)
Comment (carlo-colombo) [reply azure-om-config.html.md.erb]: did it

File: installing-pks-azure.html.md.erb (hunk 9 above)
Comment (carlo-colombo) [reply installing-pks-azure.html.md.erb]: This should be as well in the terraform, we need to ask cf-infrastructure to add it

**Pull request merged**